pkg/apis/config/v1alpha2/meshrootcertificate.go

```diff
@@ -171,8 +158,19 @@ type MeshRootCertificateStatus struct {
 	// All states are specified in constants.go
 	State string `json:"state"`
 
-	// Set of statuses for each certificate component in the cluster (e.g. webhooks, bootstrap, etc.)
-	ComponentStatuses MeshRootCertificateComponentStatuses `json:"componentStatuses"`
+	// A mapping of the component to a value describing the status of that component. The components include:
+	// - "validatingWebhook"
+	// - "mutatingWebhook"
+	// - "xdsControlPlane"
+	// - "sidecar"
+	// - "bootstrap"
+	// - "gateway"
+
+	// The values in the map can be one of (`True`, `False`, `Unknown`), True indicating the value is ready.
+	// NOTE: There is a caveat that since these components belong to horizontally scalable pods, it is possible that not
+	// all of these components will be ready. That is, one controller might mark the ADS server as ready, while all other
+	// controllers have yet to rotate their controller cert.
+	ComponentStatuses map[string]string `json:"componentStatuses"`
```

Comment:
-1 on using a map instead of strong types. Not only is this Kubernetes/Go convention, it increases the possibility of typos and inconsistent string mappings throughout the code

Reply:
Link to k8s api conventions: https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#lists-of-named-subobjects-preferred-over-maps

Reply:
good call! reverted that piece.